File: LeetCodeProblemsCMake/Problems/LeetCode2144MinimumCostOfBuyingCandiesWithDiscount.cpp
//
// Created by Saeed Jannati on 6/1/26.
//
using namespace std;
#include "LeetCode2144MinimumCostOfBuyingCandiesWithDiscount.h"

int LeetCode2144MinimumCostOfBuyingCandiesWithDiscount::minimumCost(vector<int> &cost) {
    ranges::sort(cost);
    int index{static_cast<int>(cost.size())-1};
    int spent{};
    while (index >= 0) {
        for (int i = 0; i < 2; ++i) {
            spent += cost[index--];
            if (index < 0) {
                break;
            }
        }
        --index;
    }
    return spent;
}

File: LeetCodeProblemsGo/LeetCode2144MinimumCostOfBuyingCandiesWithDiscount/LeetCode2144MinimumCostOfBuyingCandiesWithDiscount.go
package LeetCode2144MinimumCostOfBuyingCandiesWithDiscount

import "sort"

func minimumCost(cost []int) int {
	sort.Ints(cost)
	index := len(cost) - 1
	spent := 0
	for index >= 0 {
		for i := 0; i < 2; i++ {
			spent += cost[index]
			index--
			if index < 0 {
				break
			}
		}
		index--
	}
	return spent
}
